### PR TITLE
fixes #3395 - Uncaught TypeError in wpr-cpcss.js, when CPCSS buttons not displayed

### LINKED ISSUE #3292: CPCSS script is loaded on the post edit screen for non-administrator causing a JS error
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version **Yes**
- Used the search feature to ensure that the bug hasn’t been reported before **Yes**

**Describe the bug**
When you open the admin panel for a user which isn't an administrator our scripts are loaded and this makes a problem.

So for example this script

https://github.com/wp-media/wp-rocket/blob/cc3edc25191b69fa7d90ce6744c010d83415f4f8/assets/js/wpr-cpcss.js

when loaded for non-admins at edit post webpage it will give JS blocking error on `line 10` as this ID `rocket-delete-post-cpss` is not on the page (because our WP Rocket meta box is not here for this role)

**To Reproduce**
1. Open admin panel with a user has editor role.
2. Open any page/post for edit
3. Check the console so you can see an error

**Expected behavior**
All WP Rocket scripts should be loaded only for admins

**Screenshots**
image

**Additional context**
1. for now I told the customer to add the following code into his theme functions.php and this will do the trick till solving this bug (https://gist.github.com/engahmeds3ed/db6a4ea1556e262a3172dec92c4449ee)

2. We need to add the following check before adding any script to the admin page
```
if ( current_user_can( 'rocket_manage_options' ) ) { }
```

Like in the following function:
https://github.com/wp-media/wp-rocket/blob/1232eb75bf1e16a2290254af407259cdc93dee6d/inc/Engine/CriticalPath/Admin/Post.php#L99

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### LINKED ISSUE #3395: Uncaught TypeError in the wpr-cpcss.js, in case where CPCSS buttons are not displayed on some post types
**Describe the bug**
On some custom post types, such as Layout Builder in Avada theme, on the edit screen, the sidebar is not displayed/loaded. If the Optimize CSS delivery is enabled, on those custom post types that will cause the following error in the browser console: Uncaught TypeError: Cannot read property 'addEventListener' of null.

**To Reproduce**
1. Enable Optimize CSS delivery in WP Rocket
2. Go to Avada → Plugins / Add-ons → and enable Fusion Builder
3. Go to Avada → Layouts → Global Layout → Add a new section
4. Edit previously created section → open Developer tools and check the Console tab

**Expected behavior**
There should be no error in the browser console.

**Screenshots**
[https://i.imgur.com/mi5SVmM.jpeg](https://i.imgur.com/mi5SVmM.jpeg)

**Additional context**
Ticket: [https://secure.helpscout.net/conversation/1356827349/219899?folderId=3572910](https://secure.helpscout.net/conversation/1356827349/219899?folderId=3572910)

**Backlog Grooming (for WP Media dev team use only)**
- [x] Reproduce the problem
- [x] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by jorgemartine00**
## Description

Fixes #3395
Fixes #3292

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Is the solution different from the one proposed during the grooming?

The solution uses a fix on the PHP side instead of the JS side.

## How Has This Been Tested?

- [x] Added change on test site
- [x] Added change temporarily on client's site

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by remyperona** [comment]
@wp-media/php I think we should still also update the JS to avoid trying to do operations on a not existent node, what do you think?

**Comment by vg-robin** [comment]
> @wp-media/php I think we should still also update the JS to avoid trying to do operations on a not existent node, what do you think?

Any update on this? Seems there's plenty of plugins that don't show the metabox when post-editing (one called Easy Watermark brought me here), so there's still a need for those existence checks in the JavaScript.

File: inc/Engine/CriticalPath/Admin/Post.php [hunk 1]
```diff
@@ -196,6 +212,10 @@ private function get_disabled_description() {
 			return '';
 		}
 
+		if ( isset( $disabled_data['not_viewable'] ) ) {
+			return __( 'This feature is not available for not public post types.', 'rocket' );
```
Comment (iCaspar) [inline inc/Engine/CriticalPath/Admin/Post.php]: Phrasing seems a bit odd here. Maybe:
```suggestion
			return __( 'This feature is not available for non-public post types.', 'rocket' );
```

**Review by iCaspar — CHANGES_REQUESTED** [review]

**Review by iCaspar — APPROVED** [review]

**Review by Mai-Saad — APPROVED** [review]
Working as expected
[testrail-report-343.pdf](https://github.com/wp-media/wp-rocket/files/7969768/testrail-report-343.pdf)

**Pull request merged**